src/types/common.rs

```diff
@@ -751,66 +1228,139 @@ impl CryptographicDomainParameters {
+        Self::fast_scan_with(scanner, Self::TAG)
+    }
+
+    pub fn fast_scan_with(scanner: &mut FastScanner<'_>, tag: Tag) -> Result<Self, FastScanError> {
+        let mut scanner = scanner.scan_struct(tag)?;
+        let q_length = scanner.scan_opt_int(Self::Q_LENGTH_TAG)?;
+        let recommended_curve = RecommendedCurve::fast_scan_opt(&mut scanner)?;
+        scanner.finish()?;
+        Ok(Self {
+            q_length,
+            recommended_curve,
+        })
+    }
+
+    pub fn format(&self, formatter: &mut Formatter<'_>) -> FormatResult {
+        self.format_with(formatter, Self::TAG)
+    }
+
+    pub fn format_with(&self, formatter: &mut Formatter<'_>, tag: Tag) -> FormatResult {
+        let mut formatter = formatter.format_struct(tag)?;
+        if let Some(q_length) = self.q_length {
+            formatter.format_int(Self::Q_LENGTH_TAG, q_length)?;
+        }
+        if let Some(recommended_curve) = &self.recommended_curve {
+            recommended_curve.format(&mut formatter)?;
+        }
+        Ok(formatter.finish())
+    }
+}
+
 impl From<CryptographicDomainParameters> for AttributeValue {
     fn from(params: CryptographicDomainParameters) -> Self {
         AttributeValue::CryptographicDomainParameters(params.q_length, params.recommended_curve)
     }
 }
 
-/// See KMIP 1.0 section 3.14 [Cryptographic Usage Mask](https://docs.oasis-open.org/kmip/spec/v1.0/os/kmip-spec-1.0-os.html#_Toc262581188).
-// Note: This enum value is stored in a u32 but is serialized as an i32.
-#[repr(u32)]
-#[derive(EnumFlags, Clone, Copy, Deserialize, Serialize, Display, PartialEq, Eq)]
-#[rustfmt::skip]
-#[non_exhaustive]
-pub enum CryptographicUsageMask {
-    Sign                            = 0x00000001,
-    Verify                          = 0x00000002,
-    Encrypt                         = 0x00000004,
-    Decrypt                         = 0x00000008,
-    WrapKey                         = 0x00000010,
-    UnwrapKey                       = 0x00000020,
-    Export                          = 0x00000040,
-    MacGenerate                     = 0x00000080,
-    MacVerify                       = 0x00000100,
-    DeriveKey                       = 0x00000200,
-    ContentCommitmentNonRepudiation = 0x00000400,
-    KeyAgreement                    = 0x00000800,
-    CertificateSign                 = 0x00001000,
-    CrlSign                         = 0x00002000,
-    GenerateCryptogram              = 0x00004000,
-    ValidateCryptogram              = 0x00008000,
-    TranslateEncrypt                = 0x00010000,
-    TranslateDecrypt                = 0x00020000,
-    TranslateWrap                   = 0x00040000,
-    TranslateUnwrap                 = 0x00080000,
+bitflags::bitflags! {
+    /// See KMIP 1.0 section 3.14 [Cryptographic Usage Mask](https://docs.oasis-open.org/kmip/spec/v1.0/os/kmip-spec-1.0-os.html#_Toc262581188).
+    #[derive(Clone, Copy, PartialEq, Eq)]
+    pub struct CryptographicUsageMask: i32 {
```

Comment:
Why is this changed from an enum to a struct?

Reply:
I thought the whole point of bit flags is to combine them together, and in the case of a usage mask it can have multiple bits set at once. Anyway, if the reason is because of the way `bitflags` works then let's leave it like this.